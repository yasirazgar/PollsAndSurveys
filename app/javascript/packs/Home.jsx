
import React, { Component } from 'react';
import ReactDOM from 'react-dom';

import Toolbar from './Toolbar/Toolbar';
import SideDrawer from './SideDrawer/SideDrawer'
import Backdrop from './Backdrop/Backdrop'
import MainWrapper from './MainWrapper'
import Footer from './Footer/Footer'
import Loader from './Utils/Loader'
import ProfileModal from './Profile/ProfileModal'
import SignUpModal from './Session/SignUpModal'
import SignInModal from './Session/SignInModal'

import './Tooltip.scss'
import './Common.scss'

class Home extends Component {
  constructor(props){
    super(props);

    this.state = {
      sideDrawerOpen: false,
      showLoader: false,
    };
  }

  sideDrawerToggleClickHandler = () => {
    this.setState((prevState) => {
      return {sideDrawerOpen: !prevState.sideDrawerOpen};
    });
  };

  backdropClickHandler = () => {
    this.setState({sideDrawerOpen: false});
  };

  render() {
    let backdrop, loader;

    if (this.state.sideDrawerOpen) {
      this.backdrop = <Backdrop backdropClickHandler={this.backdropClickHandler} />;
    };

    if (this.state.showLoader) {
      this.loader = <Loader />
    }

    return (
      <div style={{height: '100%'}}>

        {loader}

        <Toolbar sideDrawerToggleClickHandler={this.sideDrawerToggleClickHandler} />

        <SideDrawer open={this.state.sideDrawerOpen} user={this.state.user}/>

        <MainWrapper user={this.state.user} />

        {backdrop}

        <Footer classes="normal"/>

        <SignInModal />
        <SignUpModal />
        <ProfileModal />
      </div>
    );
  }
}

export default Home
