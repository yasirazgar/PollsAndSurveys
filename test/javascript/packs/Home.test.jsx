import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Home from 'packs/Home';

const mockStore = configureMockStore();

describe("Home", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <Home />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot();
  });

  // it("sets state sideDrawerOpen to true on clicking sideDrawer toggle-button", () => {
  //   const wrapper = shallow(
  //     <Provider store={mockStore({})}>
  //       <Home />
  //     </Provider>
  //   )

  //   wrapper.find('.toggle-button').first().simulate('click');

  //   expect(wrapper.state('sideDrawerOpen')).toEqual(true);
  // });
});
