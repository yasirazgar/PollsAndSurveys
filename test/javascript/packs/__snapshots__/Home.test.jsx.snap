// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <Toolbar
    sideDrawerToggleClickHandler={[Function]}
  />
  <SideDrawer
    open={false}
  />
  ;
  <main
    style={
      Object {
        "marginTop": "113px",
      }
    }
  >
    <p>
       Main content 
    </p>
  </main>
</div>
`;
